Preparation for prod

notalex-sh opened the pull request:
## Change Summary
Modified codebase for production.

Main changes:
- Modified env files
- Modified Docker related files
- Modified django settings for static files

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [NA] The pull request title has an issue number
- [Y] The change works by "Smoke testing" or quick testing
- [NA] The change has tests
- [NA] The change has documentation

## Other Information
NO

Commit:
remove random images from server src

Commit:
remove dev container for prod, change env files for prod

Commit:
update dockerfiles and entrypoints for prod, adjust docker-compose, adjust settings.py

Alex-Hawking commented:
There is some errors with the static files as ill i havent decided on the actual path for serving on the hosting vm.

dct0 commented on docker/client/entrypoint.sh:
```diff
@@ -1,58 +1,39 @@
+elif [ "${APP_ENV}" = "DEVELOPMENT" ]; then
+    echo "Starting Next.js application in development mode"
+    npm install
+    exec npm run dev
```
Exec?

dct0 commented on .devcontainer/devcontainer.json:
Don't remove these, you can just open the repo without the dev container

dct0 commented on .devcontainer/setup.sh:
Don't remove these, you can just open the repo without the dev container

dct0 commented on docker-compose.yml:
Make a new `docker-compose-prod.yml` instead of modifying this so we have a dev and prod compose

dct0 commented on server/.env.production:
Don't delete the .env.example, create a copy. I'm not sure if this is needed though? You'll just modify the .env to the prod configuration

dct0 commented on server/api/settings.py:
```diff
@@ -155,18 +157,17 @@
+PROJECT_ROOT = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
-STATIC_URL = "/static/"
+# Will need to configure this for actual deployment
```
Can use an env var for prod

dct0 commented on client/.env.production:
Don't delete the .env.example, create a copy. I'm not sure if this is needed though? You'll just modify the .env to the prod configuration

dct0 commented on docker/client/Dockerfile:
```diff
@@ -21,5 +18,3 @@ COPY ./client /app
 RUN chmod +x /entrypoint.sh
 CMD ["/entrypoint.sh"]
-
-# TODO: Production 
```
Typically a prod build would have nextjs build as "standalone" and that would be then copied inside the container, but this should be fine for most purposes

dct0 reviewed (changes requested):
Nice work! Few comments relating from the dev perspective

harryrigg commented on server/api/settings.py:
```diff
@@ -176,6 +177,6 @@
-    # "ACCESS_TOKEN_LIFETIME": datetime.timedelta(seconds=10),
-    # "REFRESH_TOKEN_LIFETIME": datetime.timedelta(seconds=30),
-}
+    "ACCESS_TOKEN_LIFETIME": datetime.timedelta(seconds=10),
+    "REFRESH_TOKEN_LIFETIME": datetime.timedelta(seconds=30),
```
Remove these lines as it will cause user session to expire in 30 seconds. This is just a remnant from when I was trying to fix the token refresh.

CodeWithTheDoctor committed:
Restore devcontainer.json file

CodeWithTheDoctor committed:
- readded setup.sh

CodeWithTheDoctor committed:
Reverted docker compose dev

CodeWithTheDoctor committed:
- Addressed recommendations from PR review:
  - Added env variable from project root in django settings.py
  - Brought .env.examples back to life

CodeWithTheDoctor committed:
In progress: Making a prod-ready compose and compose files.
Updated from django files as per Yunho's recs to fix some of the issues occuring while building

CodeWithTheDoctor committed:
- Made progress to prod deployment configs.
Frontend deploys successfully!
Backend in progress: Trying to get it to createsuperuser. command doesnt work manually in the container on vps atm.

CodeWithTheDoctor committed:
Trying to get this working
